internal/pkg/testing/esutil/client.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package esutil
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/stretchr/testify/require"
+)
+
+/*
+	Setup to convert a *elasticsearch.Client as a harmless mock
+	by replacing the Transport to nowhere
+*/
+
+type MockTransport struct {
+	Response    *http.Response
+	RoundTripFn func(req *http.Request) (*http.Response, error)
+}
+
+func (t *MockTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	return t.RoundTripFn(req)
+}
+
+func MockESClient(t *testing.T) (*elasticsearch.Client, *MockTransport) {
```

Comment:
copied this mock from [here](https://github.com/elastic/fleet-server/blob/a465d093a882d8b4a9ec1ae97d8494b0d2f9ae39/internal/pkg/api/handleUpload_test.go#L1112) to be accessible from another package